cfme/web_ui/expression_editor_widgetastic.py

```diff
@@ -8,12 +8,71 @@
 import re
 from utils.pretty import Pretty
 
-from widgetastic_patternfly import Input, BootstrapSelect
-from widgetastic.widget import Widget, View
+from widgetastic_patternfly import Input, BootstrapSelect, Button
+from widgetastic.widget import View
 from widgetastic_manageiq import Calendar
+from widgetastic.utils import VersionPick, Version
 
 
-class ExpressionEditor(Widget, Pretty):
+class FieldFormView(View):
+    type = BootstrapSelect("chosen_typ")
+    field = BootstrapSelect("chosen_field")
+    key = BootstrapSelect("chosen_key")
+    value = Input("chosen_value")
+    user_input = Input("user_input")
+
+
+class FieldDateForm(View):
+    dropdown_select = BootstrapSelect("chosen_from_1")
+    input_select_date = Calendar("miq_date_1_0")
+    input_select_time = BootstrapSelect("miq_time_1_0")
+
+
+class CountFormView(View):
+    type = BootstrapSelect("chosen_typ")
+    count = BootstrapSelect("chosen_count")
+    key = BootstrapSelect("chosen_key")
+    value = Input("chosen_value")
+    user_input = Input("user_input")
+
+
+class TagFormView(View):
+    type = BootstrapSelect("chosen_typ")
+    tag = BootstrapSelect("chosen_tag")
+    value = BootstrapSelect("chosen_value")
+    user_input = Input("user_input")
+
+
+class FindFormView(View):
+    type = BootstrapSelect("chosen_typ")
+    field = BootstrapSelect("chosen_field")
+    skey = BootstrapSelect("chosen_skey")
+    value = Input("#chosen_value")
+    check = BootstrapSelect("chosen_check")
+    cfield = BootstrapSelect("chosen_cfield")
+    ckey = BootstrapSelect("chosen_ckey")
+    cvalue = Input("chosen_cvalue")
+
+
+class RegistryFormView(View):
+    type = BootstrapSelect("chosen_typ")
+    key = Input("chosen_regkey")
+    value = Input("chosen_regval")
+    operation = BootstrapSelect("chosen_key")
+    contents = Input("chosen_value")
+
+
+class DateSpecificForm(View):
```

Comment:
Two things:

1. Naming consistency - ThingFormView, ThingFormView, ThingForm - make them all one or the other, I'm partial to calling them 'Form'
2. These appear to be nested views - why define these outside of the ExpressionEditor, and then set class attributes for each? This treats them like widgets instead of like nested views. This may be correct, I'm still learning, @mfalesni should probably weigh in on this.